core/ui/src/main/java/com/google/samples/apps/nowinandroid/core/ui/NewsResourceCard.kt

```diff
@@ -228,28 +230,40 @@ fun NewsResourceShortDescription(
 
 @Composable
 fun NewsResourceTopics(
-    topics: List<Topic>,
+    topics: List<FollowableTopic>,
     modifier: Modifier = Modifier
 ) {
     // Store the ID of the Topic which has its "following" menu expanded, if any.
     // To avoid UI confusion, only one topic can have an expanded menu at a time.
     var expandedTopicId by remember { mutableStateOf<String?>(null) }
+    val isFollowed = stringResource(R.string.topic_is_followed)
+    val isNotFollowed = stringResource(R.string.topic_is_not_followed)
 
     Row(
         modifier = modifier.horizontalScroll(rememberScrollState()), // causes narrow chips
         horizontalArrangement = Arrangement.spacedBy(4.dp),
     ) {
-        for (topic in topics) {
+        for (followableTopic in topics) {
             NiaTopicTag(
-                expanded = expandedTopicId == topic.id,
-                followed = true, // ToDo: Check if topic is followed
+                expanded = expandedTopicId == followableTopic.topic.id,
+                followed = followableTopic.isFollowed,
                 onDropdownMenuToggle = { show ->
-                    expandedTopicId = if (show) topic.id else null
+                    expandedTopicId = if (show) followableTopic.topic.id else null
                 },
                 onFollowClick = { }, // ToDo
                 onUnfollowClick = { }, // ToDo
                 onBrowseClick = { }, // ToDo
-                text = { Text(text = topic.name.uppercase(Locale.getDefault())) }
+                text = {
+                    Text(
+                        text = followableTopic.topic.name.uppercase(Locale.getDefault()),
+                        modifier = Modifier.semantics {
+
+                            contentDescription = followableTopic.topic.name + " " +
```

Comment:
There's a better solution here using string substitution. I will push it so you can see what I mean.